src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -473,6 +482,17 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
                     var call = (BoundCall)expr;
                     return CheckCallValueKind(call, node, valueKind, checkingReceiver, diagnostics);
 
+                case BoundKind.IndexOrRangePatternIndexerAccess:
+                    var patternIndexer = (BoundIndexOrRangePatternIndexerAccess)expr;
+                    // Index access should have been handled with properties, above
```

Comment:
Nit: assert that's true?

Reply:
Below I explicitly convert to MethodSymbol, which I think is an even stronger assertion.